File: Aliucord/src/main/java/com/aliucord/views/ToolbarButton.java
package com.aliucord.views;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.ContextThemeWrapper;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatImageButton;

import com.aliucord.Utils;
import com.lytefast.flexinput.R;

/** Settings Header Toolbar Button */
public class ToolbarButton extends AppCompatImageButton {
    /**
     * Creates a header-toolbar button
     * @param context {@link Context}
     */
    public ToolbarButton(Context context) {
        super(new ContextThemeWrapper(context, R.h.UiKit_ImageView_Clickable), null, 0);
    }

    @Override
    public void setImageDrawable(@Nullable Drawable drawable) {
        setImageDrawable(drawable, true);
    }

    public void setImageDrawable(@Nullable Drawable drawable, boolean forceTint) {
        if (forceTint && drawable != null) {
            drawable = drawable.mutate();
            Utils.tintToTheme(getContext(), drawable);
        }
        super.setImageDrawable(drawable);
    }
}

File: Aliucord/src/main/java/com/aliucord/views/ToolbarButton.kt
package com.aliucord.views

import android.content.Context
import android.graphics.drawable.Drawable
import android.view.ContextThemeWrapper
import androidx.appcompat.widget.AppCompatImageButton
import com.aliucord.Utils
import com.lytefast.flexinput.R

class ToolbarButton(context: Context) :
    AppCompatImageButton(ContextThemeWrapper(context, R.i.UiKit_ImageView_Clickable), null, 0) {
    override fun setImageDrawable(drawable: Drawable?) {
        setImageDrawable(drawable, true)
    }

    fun setImageDrawable(drawable: Drawable?, forceTint: Boolean) {
        if (forceTint && drawable != null) {
            val drawable = drawable.mutate()
            Utils.tintToTheme(drawable)
        }
        super.setImageDrawable(drawable)
    }
}
